expression/aggregation/bit_or.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregation
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
+	"github.com/pingcap/tidb/types"
+)
+
+type bitOrFunction struct {
+	aggFunction
+}
+
+// Clone implements Aggregation interface.
+func (bf *bitOrFunction) Clone() Aggregation {
+	nf := *bf
+	for i, arg := range bf.Args {
+		nf.Args[i] = arg.Clone()
+	}
+	return &nf
+}
+
+// CalculateDefaultValue implements Aggregation interface.
+func (bf *bitOrFunction) CalculateDefaultValue(schema *expression.Schema, ctx context.Context) (d types.Datum, valid bool) {
+	arg := bf.Args[0]
```

Comment:
Should we also check len(bf.Args[0]) as line 62?

Reply:
@shenli actually I think the check in line `62` should be removed as well. There's no need check argument length everytime `Update` is called.  Parser would fail if argument number is wrong unless something is wrong in the framework.  Or we can do this check in `expression/aggregation/aggregation.go`